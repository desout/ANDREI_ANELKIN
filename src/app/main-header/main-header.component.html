<nav class="navbar navbar-expand-md navbar-dark fixed-top bg-dark">
  <ul class="navbar-nav mr-auto">
    <li>
      <span class="navbar-brand">{{title}}</span>
    </li>
    <li *ngIf="(userService.currentUser$ | async)?.name" class="nav-item mr-2">
      <a class="nav-link p-2 text-light" routerLink="/infoTab" routerLinkActive="active">{{'HOME.NAVINFO' | translate}} </a>
    </li>
    <li *ngIf="(userService.currentUser$ | async)?.name" class="nav-item mr-2">
      <a class="nav-link p-2 text-light" routerLink="/editTab" routerLinkActive="active">{{'HOME.NAVEDIT' | translate}}</a>
    </li>
    <li *ngIf="(userService.currentUser$ | async)?.role === 'ADMIN'" class="nav-item mr-2">
      <a class="nav-link p-2 text-light" routerLink="/adminTab" routerLinkActive="active">{{'HOME.NAVADMIN' | translate}}</a>
    </li>
  </ul>
  <ul class="navbar-nav ml-auto">
    <li *ngIf="(userService.currentUser$ | async)?.name" class="nav-item mr-3">
      <span class="nav-link p-2 text-light">{{'HOME.LOGGERHEAD' | translate}} {{(userService.currentUser$ | async)?.name}}</span>

    </li>
    <li *ngIf="(userService.currentUser$ | async)?.name" class="nav-item mr-3">
      <a class="btn btn-outline-warning text-light" (click)="onClickLogout()">{{'HOME.NAVLOGOUT' | translate}}</a>
    </li>
    <li class="nav-item">
      <label class="nav-link">
        {{ 'HOME.SELECT' | translate }}
        <select #langSelect (change)="translate.use(langSelect.value)">
          <option *ngFor="let lang of translate.getLangs()" [value]="lang" [selected]="lang === translate.currentLang">{{ lang }}</option>
        </select>
      </label>
    </li>
  </ul>
</nav>
