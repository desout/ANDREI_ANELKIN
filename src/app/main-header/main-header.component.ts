import {Component, OnInit} from '@angular/core';
import {TranslateService} from '@ngx-translate/core';
import {UserService} from '../user.service';
import {AuthService} from '../auth.service';
import {LocalUser} from '../../../server/server/LocalUser';

@Component({
  selector: 'app-main-header',
  templateUrl: './main-header.component.html',
  styleUrls: ['./main-header.component.scss']
})
export class MainHeaderComponent implements OnInit {
  title = 'Andrei Anelkin';
  user: LocalUser;
  constructor(public translate: TranslateService, private userService: UserService, private authService: AuthService) {

    translate.addLangs(['en', 'ru']);
    translate.setDefaultLang('en');
    translate.use('en');
  }
  onClickLogout() {
    this.authService.logout();
  }
  ngOnInit() {
    this.userService.userUpdateHandle
      .subscribe((letUser) => this.user = letUser);
  }

}
