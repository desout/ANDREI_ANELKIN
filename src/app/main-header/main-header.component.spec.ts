import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { MainHeaderComponent } from './main-header.component';
import {NO_ERRORS_SCHEMA} from '@angular/core';
import {TestModule} from '../Test.module';

describe('MainHeaderComponent', () => {
  let component: MainHeaderComponent;
  let fixture: ComponentFixture<MainHeaderComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [],
      schemas: [NO_ERRORS_SCHEMA],
      imports: [TestModule],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(MainHeaderComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it( 'should be select En lang', () => {
    const select: HTMLSelectElement = fixture.nativeElement.querySelector('select');

    expect(select.value).toEqual('en');
    expect(select.value).toMatchSnapshot();
  });
});
