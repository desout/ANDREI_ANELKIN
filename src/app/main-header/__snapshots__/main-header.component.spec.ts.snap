// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainHeaderComponent should be select En lang 1`] = `"en"`;

exports[`MainHeaderComponent should create 1`] = `
<div
  id="root0"
  ng-version="7.0.3"
>
  <nav
    class="navbar navbar-expand-md navbar-dark fixed-top bg-dark"
  >
    <ul
      class="navbar-nav mr-auto"
    >
      <li>
        <span
          class="navbar-brand"
        >
          Andrei Anelkin
        </span>
      </li>
      
      
      
    </ul>
    <ul
      class="navbar-nav ml-auto"
    >
      
      
      <li
        class="nav-item"
      >
        <label
          class="nav-link"
        >
            
          <select>
            
            <option
              ng-reflect-value="en"
              value="en"
            >
              en
            </option>
            <option
              ng-reflect-value="ru"
              value="ru"
            >
              ru
            </option>
          </select>
        </label>
      </li>
    </ul>
  </nav>
</div>
`;
